File: lib/dc/language.rb
module DC
  # The official list of supported languages

  # should we be using iso15924 script codes?
  #   http://www.unicode.org/iso15924/iso15924-codes.html
  module Language
    NAMES = {
      'eng' => 'English',
      'spa' => 'Spanish',
      'fra' => 'French',
      'deu' => 'German',
      'ita' => 'Italiano'
    }
    SUPPORTED = NAMES.keys
    DEFAULT = 'ita'

    USER = ['ita','spa','eng']
  end

end

File: public/javascripts/model/language.js
// Keep this file in sync with lib/dc/language.rb

dc.language = {
  NAMES: {
    'eng': 'English',
    'spa': 'Español/Spanish',
    'fra': 'Français/French',
    'deu': 'Duetch/German',
    'ita': 'Italiano'

  },
  USER: ['eng','spa','ita']
};

dc.language.SUPPORTED = _.keys(dc.language.NAMES);
